javascript/javascript1/week2/smartease.js

```diff
@@ -0,0 +1,177 @@
+// Flight booking fullname function
+console.log("**********************");
+console.log("Ex 1 : Flight booking fullname function");
+console.log("**********************");
+
+function getFullName(firstname, surname) {
+  return firstname + " " + surname;
+}
+const fullName1 = getFullName("Javascript", "class");
+const fullName2 = getFullName("React", "class");
+console.log(`fullName1:${fullName1}\nfullName2:${fullName2}`);
+
+//Formal fullname
+console.log("**********************");
+console.log("Ex 2: Formal fullname");
+console.log("**********************");
+function getFormalFullName(firstname, surname, useFormalName, gender) {
+  if (!firstname || !surname) {
+    return "Please provide your firstname and surname";
+  } else {
+    if (useFormalName === false || !useFormalName) {
+      return `${firstname} ${surname}`;
+    } else if (typeof useFormalName !== "boolean") {
+      return "please mention useFormalName as true or false";
+    } else if (useFormalName === true && gender === "male") {
+      return `Lord ${firstname} ${surname}`;
+    } else if (useFormalName === true && gender === "female") {
+      return `Lady ${firstname} ${surname}`;
+    } else {
+      return "Please provide the gender";
+    }
+  }
+}
+const formalFullName1 = getFormalFullName("Benjamin", "Hughes", false);
+const formalFullName2 = getFormalFullName("Ida", "Jones");
+console.log(
+  `formalFullName1:${formalFullName1}\nformalFullName2:${formalFullName2}`
+);
+
+//Event application
+console.log("**********************");
+console.log("Ex 3: Event application");
+console.log("**********************");
+
+function getEventWeekday(daysAfter) {
+  const today = new Date().getDay();
+  const weekDays = [
+    "Sunday",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+
+  return weekDays[(today + daysAfter) % 7];
+}
+const daysAfter = 4;
+const eventDay = getEventWeekday(daysAfter);
+console.log(eventDay);
+
+//Weather wear
+console.log("**********************");
+console.log("Ex 4: Weather wear");
+console.log("**********************");
+
+function selectClothes(temperature) {
+  if (temperature <= 5) {
+    return "Winter Jacket and Thermal Wear";
+  } else if (temperature > 5 && temperature < 18) {
+    return "Normal Jacket and Pant and T-shirt";
+  } else if (temperature >= 18) {
+    return "Shorts and a T-shirt";
+  } else {
+    return "Please provide the current temperature in degrees";
+  }
+}
+const temperature = 18;
+const clothesToWear = selectClothes(temperature);
+console.log(clothesToWear);
```

Comment:
in the first `else if` you do not have to check `temperature > 5` as temperature must be large than 5, otherwise the first `if` would have kicked in.

Reply:
Yes, I agree with that and I have changed the code.